java/vector/src/main/codegen/templates/UnionVector.java

```diff
@@ -647,6 +673,8 @@ public ValueVector getVectorByType(int typeId, ArrowType arrowType) {
           return getStruct();
         case LIST:
           return getList();
+        case MAP:
+          return getMap();
```

Comment:
here we should call `getMap(arrowType)`?

Reply:
I wonder if it should be `getMap(name, arrowType)` just like decimal

Reply:
I believe this should've been `getMap(name, arrowType)`. I updated the PR with a unit test for `UnionVector`.